src/NClient/NClient.Standalone/Client/Host/CompositeHost.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NClient.Providers.Host;
+
+namespace NClient.Standalone.Client.Host
+{
+    internal class CompositeHost : IHost
+    {
+        private readonly IReadOnlyCollection<IHost> _hosts;
+        
+        public CompositeHost(IReadOnlyCollection<IHost> hosts)
+        {
+            _hosts = hosts;
+        }
+        
+        public async Task<Uri?> TryGetUriAsync(CancellationToken cancellationToken)
+        {
+            foreach (var host in _hosts)
+            {
+                var uri = await host.TryGetUriAsync(cancellationToken).ConfigureAwait(false);
+                if (uri is not null)
+                    return uri;
```

Comment:
It turns out that no matter how many times we call WithHost, the first host will always be used?

Reply:
Right now ... yes. I don't see any relevant cases for many hosts without load-balance feature etc, but if we'll implement service discovery and load-balance, this logic will be rewrite.

Reply:
I suggest using not the first, but the last set host :)

Reply:
As you wish.